.editorconfig

```diff
@@ -1,40 +1,46 @@
-# editorconfig.org
-
-# top-most EditorConfig file
-root = true
-
-# Default settings:
-# A newline ending every file
-# Use 4 spaces as indentation
-[*]
-insert_final_newline = true
-end_of_line = crlf
-indent_style = space
-indent_size = 4
-
-# Trim trailing whitespace, limited support.
-# https://github.com/editorconfig/editorconfig/wiki/Property-research:-Trim-trailing-spaces
-trim_trailing_whitespace = true
-
-[*.{cs,vb}]
-dotnet_style_predefined_type_for_locals_parameters_members = true:error
-
-dotnet_naming_rule.private_members_with_underscore.symbols  = private_fields
-dotnet_naming_rule.private_members_with_underscore.style    = prefix_underscore
-dotnet_naming_rule.private_members_with_underscore.severity = suggestion
-
-dotnet_naming_symbols.private_fields.applicable_kinds           = field
-dotnet_naming_symbols.private_fields.applicable_accessibilities = private
-
-dotnet_naming_style.prefix_underscore.capitalization = camel_case
-dotnet_naming_style.prefix_underscore.required_prefix = _
-
-# https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/ide/editorconfig-code-style-settings-reference.md
-[*.cs]
-csharp_style_var_for_built_in_types = true:suggestion
-csharp_style_var_when_type_is_apparent = true:suggestion
-csharp_style_var_elsewhere = true:suggestion
+# editorconfig.org
+
+# top-most EditorConfig file
+root = true
+
+# Default settings:
+# A newline ending every file
+# Use 4 spaces as indentation
+[*]
+insert_final_newline = true
+end_of_line = crlf
+indent_style = space
+indent_size = 4
+
+# Trim trailing whitespace, limited support.
+# https://github.com/editorconfig/editorconfig/wiki/Property-research:-Trim-trailing-spaces
+trim_trailing_whitespace = true
+
+[*.{cs,vb}]
+dotnet_style_predefined_type_for_locals_parameters_members = true:error
+
+dotnet_naming_rule.private_members_with_underscore.symbols  = private_fields
+dotnet_naming_rule.private_members_with_underscore.style    = prefix_underscore
+dotnet_naming_rule.private_members_with_underscore.severity = suggestion
+
+dotnet_naming_symbols.private_fields.applicable_kinds           = field
+dotnet_naming_symbols.private_fields.applicable_accessibilities = private
+# dotnet_naming_symbols.private_fields.required_modifiers = abstract,async,readonly,static # all except const
+
+dotnet_naming_style.prefix_underscore.capitalization = camel_case
+dotnet_naming_style.prefix_underscore.required_prefix = _
+
+
+
+
+
+
+# https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/ide/editorconfig-code-style-settings-reference.md
+[*.cs]
+csharp_style_var_for_built_in_types = true:suggestion
+csharp_style_var_when_type_is_apparent = true:suggestion
+csharp_style_var_elsewhere = true:suggestion
 csharp_prefer_braces = false : none

```

Comment:
This is such a dangerous setting. 

https://embeddedgurus.com/barr-code/2014/03/apples-gotofail-ssl-security-bug-was-easily-preventable/

Reply:
Was added here https://github.com/umbraco/Umbraco-CMS/commit/1d9bb4605bba814d8954b204f1dd5ea099680031 I assume because many moons ago when v8 was being developed by community members if statements with braces were being commonly used. I've been stung with almost the same problem because of this. This rule would also allow for things like if/foreach without any braces... which is far too nasty. 

I think at the time of this rule/development r# was heavily relied upon which would warn of these mistakes. I don't use r# anymore for various reason and I know plenty of others don't either so we can't rely on that. 

I am a fan of terseness but that is only my personal opinion in my own code. I agree that its much safer to be consistent and require braces since there doesn't seem to be a rule that would prevent those types of mistakes.

Reply:
@Shazwazza what's the resolution here? Is this cool to merge with prefer false?

Reply:
@nathanwoulfe we should align whatever is in the codebase in v8 with what is in netcore since that is what we will be doing from here on in: 

So we shouldn't merge these at all IMO unless they are inline with what is in the netcore proj: https://github.com/umbraco/Umbraco-CMS/blob/netcore/netcore/src/.editorconfig